samples/snippets/src/main/java/com/example/storage/object/DeleteObject.java

```diff
@@ -38,16 +39,18 @@ public static void deleteObject(String projectId, String bucketName, String obje
       System.out.println("The object " + objectName + " wasn't found in " + bucketName);
       return;
     }
-
-    // Optional: set a generation-match precondition to avoid potential race
-    // conditions and data corruptions. The request to upload returns a 412 error if
-    // the object's generation number does not match your precondition.
-    Storage.BlobSourceOption precondition =
-        Storage.BlobSourceOption.generationMatch(blob.getGeneration());
-
-    storage.delete(bucketName, objectName, precondition);
-
-    System.out.println("Object " + objectName + " was deleted from " + bucketName);
+    BlobId idWithGeneration = blob.getBlobId();
+    // Deletes the blob specified by its id. When the generation is present and non-null it will be
```

Comment:
I think "when the generation is present" is a little vague, maybe we should specify as "When the generation is present in the BlobId object"?

Reply:
rephrased the comments. let me know if that is clearer.